src/JavaCodeGenerator.cs

```diff
@@ -1818,107 +1687,61 @@ private static ServiceModelProperty ParseModelProperty(AutoRestProperty property
 
             List<string> annotationArgumentList = new List<string>()
             {
-                $"value = \"{(settings.ShouldGenerateXmlSerialization ? property.XmlName : property.SerializedName)}\""
+                $"value = \"{(settings.ShouldGenerateXmlSerialization ? autoRestProperty.XmlName : autoRestProperty.SerializedName)}\""
             };
-            if (property.IsRequired)
+            if (autoRestProperty.IsRequired)
             {
                 annotationArgumentList.Add("required = true");
             }
-            if (property.IsReadOnly)
+            if (autoRestProperty.IsReadOnly)
             {
                 annotationArgumentList.Add("access = JsonProperty.Access.WRITE_ONLY");
             }
             string annotationArguments = string.Join(", ", annotationArgumentList);
 
-            bool isXmlAttribute = property.XmlIsAttribute;
+            bool isXmlAttribute = autoRestProperty.XmlIsAttribute;
 
             string xmlName;
             try
             {
-                xmlName = property.XmlName;
+                xmlName = autoRestProperty.XmlName;
             }
             catch (ArgumentNullException)
             {
                 xmlName = null;
             }
 
-            string serializedName = property.SerializedName;
+            string serializedName = autoRestProperty.SerializedName;
 
-            bool isXmlWrapper = property.XmlIsWrapped;
+            bool isXmlWrapper = autoRestProperty.XmlIsWrapped;
 
-            AutoRestIModelType propertyModelType = property.ModelType;
-            AutoRestPrimaryType propertyModelPrimaryType = propertyModelType as AutoRestPrimaryType;
-            if (propertyModelType != null && !IsNullable(property) && propertyModelPrimaryType != null)
+            AutoRestIModelType autoRestPropertyModelType = autoRestProperty.ModelType;
+            IType propertyWireType = ParseType(autoRestPropertyModelType, settings);
+            if (IsNullable(autoRestProperty))
             {
-                AutoRestPrimaryType propertyModelNonNullablePrimaryType = DependencyInjection.New<AutoRestPrimaryType>(propertyModelPrimaryType.KnownPrimaryType);
-                propertyModelNonNullablePrimaryType.Format = propertyModelPrimaryType.Format;
-                primaryTypeNotWantNullable.Add(propertyModelNonNullablePrimaryType);
-
-                propertyModelType = propertyModelNonNullablePrimaryType;
+                propertyWireType = propertyWireType.AsNullable();
             }
+            IType propertyClientType = ConvertToClientType(propertyWireType);
 
-            string xmlListElementName = propertyModelType is AutoRestSequenceType st ? st.ElementXmlName : null;
+            string xmlListElementName = autoRestPropertyModelType is AutoRestSequenceType autoRestPropertyModelSequenceType ? autoRestPropertyModelSequenceType.ElementXmlName : null;
```

Comment:
It's not more informative to use a full name here because we know it's the same object instance

Reply:
Referring to `st`-- also not a big deal either way